@here/harp-mapview/lib/FrustumIntersection.ts

```diff
@@ -120,16 +123,21 @@ export class FrustumIntersection {
             }
 
             const tileKey = tileEntry.tileKey;
+            if (tileKey.level > maxTileLevel) {
```

Comment:
Why not just have a condition, i
`if (tileKey.level === maxTileLevel) {
continue;
}`
I don't understand the need for iterating over the datasources.

Reply:
@atomicsulfate , you also worked on this? Is this acceptable?

Reply:
@nzjony, this enables custom culling on datasources, defined not only by zoomlevel.
For example, in poles datasource you don't ever want to check tiles outside of pole radius,
https://github.com/heremaps/harp.gl/pull/653/files#diff-1b0ff12045051a049d60bf7d1c58e4a3R140

Reply:
besides, only this change reduces overall subdivisions per frame. I didn't check why, but it is

Reply:
Shouldn't this then be a property of the TilingScheme and not the DataSource? I guess the poles have some custom TilingScheme, i think it makes more sense to have it there no?

Reply:
No.
For polar datasource, the transverse Mercator projection is used along with regular quadtree. With that scheme, all the spherical space except two regions at (0, -90), (0, +90) is defined and can be addressed. And for pole datasource, it is vital only to have north and south pole addressed (+90, 0), (-90, 0) with a radius of ~5 degrees.
So, although it sounds like a good idea, this can't be defined in terms of TilingScheme

Reply:
Just add the method to TilingScheme, and have a PolarTilingScheme inherit from the TransverseMercatorProjection?
Would make more sense that having the DataSource knowing if it should subdivide or not.

Reply:
It's a bad idea because we can start to spam tiling schemas here and there, while we actually want to reduce them due to high traverse complexity.
If in other way you'd have two equal schemas traversed once, with this approach you'll have two different schemas traversed twice.

Reply:
I don't quite follow, you have a polarTilingScheme:

```
export const polarTilingScheme = new TilingScheme(
    quadTreeSubdivisionScheme,
    transverseMercatorProjection
);
```

This will be treated separately in the VisibleTileSet, so you anyway have to traverse it separately from the other TilingSchemes, why would specializing this further cause a double traversal?

Reply:
You can add another datasource with same tiling scheme, but different clipping rules, and traversing would occur only once. But I see your point, if we skip some tiles completely, wouldn't it be different tiling scheme? May be. It's a topic for discussion and testing

Reply:
@nzjony Having the shouldSubdivide in the DataSource is much more flexible as having it in the tiling scheme. Imagine a usecase where you depending on the data in a tile dynamically decide that you don't want to subdivide any further. This could not be done if the shouldSubdivide is in the TilingScheme

Reply:
All good, I anyway approved these changes on the 5th of August, so I don't know why it wasn't submitted.